pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4850,54 +4851,51 @@ cpdef to_offset(freq, bint is_period=False):
-                        f"in a future version, please use "
-                        f"\'{name.upper()}\' instead.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
-                    )
-                    name = name.upper()
-                if is_period and name.upper() in c_REVERSE_OFFSET_DEPR_FREQSTR:
-                    if name.upper().startswith("Y"):
+                if not is_period:
+                    if name.upper() in c_OFFSET_REMOVED_FREQSTR:
                         raise ValueError(
-                            f"for Period, please use \'Y{name.upper()[2:]}\' "
-                            f"instead of \'{name}\'"
+                            f"\'{name}\' is no longer supported for offsets. Please "
+                            f"use \'{c_OFFSET_REMOVED_FREQSTR.get(name.upper())}\' "
+                            f"instead."
                         )
+                    if (name.upper() != name and
+                            name.lower() not in {"W-TUE", "s", "ms", "us", "ns"} and
+                            name.upper().split("-")[0].endswith(("S", "E"))):
+                        raise ValueError(INVALID_FREQ_ERR_MSG.format(name))
+                if (is_period and
+                        name.upper() in c_OFFSET_TO_PERIOD_FREQSTR and
+                        name != "ms" and
+                        name.upper().split("-")[0].endswith(("S", "E"))):
                     if (name.upper().startswith("B") or
                             name.upper().startswith("S") or
                             name.upper().startswith("C")):
                         raise ValueError(INVALID_FREQ_ERR_MSG.format(name))
                     else:
+                        name_msg = "".join(name.upper().split("E", 1))
                         raise ValueError(
-                            f"for Period, please use "
-                            f"\'{c_REVERSE_OFFSET_DEPR_FREQSTR.get(name.upper())}\' "
+                            f"for Period, please use \'{name_msg}\' "
                             f"instead of \'{name}\'"
                         )
-                elif is_period and name.upper() in c_OFFSET_DEPR_FREQSTR:
-                    if name.upper() != name:
+                if is_period:
+                    if name.upper() in c_OFFSET_REMOVED_FREQSTR:
+                        if name.upper() != name:
+                            raise ValueError(
+                                f"\'{name}\' is no longer supported, "
+                                f"please use \'{name.upper()}\' instead.",
+                            )
+                        name = c_OFFSET_REMOVED_FREQSTR.get(name.upper())
+
+                    if ((name.upper() in c_PERIOD_AND_OFFSET_ALIASES and
+                            name.upper() != name) or
+                            name == "MIN"):
+                        name_new = "min" if name == "MIN" else name.upper()
```

Comment:
I don't understand this - is it  possible to make multiple dictionaries to make it cleaner?

Reply:
yes, I added a dictionary` c_PERIOD_AND_OFFSET_DEPR_FREQSTR`, to store frequencies which are the same for offsets and period, like: '`W', 'min'`. We deprecate wrong casing for these frequencies, such as: `'w', 'MIN'.` As a follow up we will add to this dictionaryfrequencies:`'D', 'C', 'B'.`